webapp/components/NewsList/NewsList.spec.js

```diff
@@ -1,75 +1,130 @@
-const ascendingTestItems = [
-    { id: 3, title: "React", votes: 0 },
-    { id: 1, title: "VueJS", votes: 1 },
-    { id: 2, title: "TDD", votes: 4 },
-];
+const postListMock = {
+    data: {
+        posts: [
+            {
+                id: "1",
+                title: "Vue",
+                votes: 4,
+                author: {
+                    id: "1",
+                    __typename: "User",
+                },
+                __typename: "Post",
+            },
+            {
+                id: "2",
+                title: "React",
+                votes: 0,
+                author: {
+                    id: "2",
+                    __typename: "User",
+                },
+                __typename: "Post",
+            },
+            {
+                id: "3",
+                title: "TDD",
+                votes: 2,
+                author: {
+                    __typename: "User",
+                    id: "3",
+                },
+                __typename: "Post",
+            },
+        ],
+    },
+};
+
+const localVue = createLocalVue();
+localVue.use(VueApollo);
+localVue.use(Vuex);
 
 describe("NewsList.vue", () => {
-    describe("empty", () => {
-        it("renders a message when the item list is empty", () => {
-            const wrapper = shallowMount(NewsList, {
-                propsData: {
-                    initialItems: [],
+    let wrapper;
+    let mockClient;
+    let apolloProvider;
+    let requestHandlers;
+    let store = new Vuex.Store({
+        getters: {
+            loggedIn: () => false,
+        },
+        mutations: {
+            setToken() {},
+        },
+    });
+
+    const createComponent = (handlers) => {
+        mockClient = createMockClient({
+            resolvers: {},
+        });
+        requestHandlers = {
+            allPostsQueryHandler: jest.fn().mockResolvedValue({ ...postListMock }),
+            ...handlers,
+        };
+        mockClient.setRequestHandler(allPostsQuery, requestHandlers.allPostsQueryHandler);
+        apolloProvider = new VueApollo({ defaultClient: mockClient });
+        const getToken = jest.fn();
+        wrapper = mount(NewsList, {
+            store,
+            localVue,
+            apolloProvider,
+            mocks: {
+                $apolloHelpers: {
+                    getToken,
                 },
+            },
+        });
+    };
+
+    afterEach(() => {
+        wrapper.destroy();
+        mockClient = null;
+        apolloProvider = null;
+    });
+
+    it("renders a Vue component", () => {
+        createComponent();
+        expect(wrapper.exists()).toBe(true);
+        expect(wrapper.vm.$apollo.queries.items).toBeTruthy();
+    });
+
+    describe("empty", () => {
+        it("renders a message when the item list is empty", async () => {
+            createComponent({
+                allPostsQueryHandler: jest.fn().mockResolvedValue({ data: { posts: [] } }),
             });
+            await wrapper.vm.$nextTick();
+
```

Comment:
⭐ Refactor looks good. We also used `wrapper.vm.$nextTick()` which is considered ugly but at least it works.